src/index.js

```diff
@@ -113,7 +115,7 @@ export default function({ template, types, traverse }) {
               plugin = plugin.default
             }
 
-            return plugin({ template, types }).visitor
+            return plugin(api).visitor
           })
 
           traverse(
```

Comment:
It is also weird that you re-use `pluginsState` here, because plugins generally assume that they each have a unique `state` instance, so this could cause plugins that have common names for internal state flags to corrupt eachother, not to mention that the options they see are the same options passed to this plugin, which means if we add an option that happens to overlap with this plugin's options, it could get turned on automatically and break things for users unexpectedly.

Reply:
This pull-request main focus was to upgrade the babel dependencies to the version Next.js is relying on.

Reply:
Changing the order in the configuration didn't help. We traverse the tree up-front to prevent being applied after the env preset:
https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types/blob/517bc5107993d738001438235c27b05cfc246969/src/index.js#L130-L131

Reply:
Maybe the situation has changed now. Back in the time, and with my limited knoledge of Babel, it was the best solution I could came up with.